src/HttpRequest.cc

```diff
@@ -721,6 +723,43 @@ HttpRequest::parseHeader(const char *buffer, const size_t size)
     return header.parse(buffer, size, clen);
 }
 
+void
+HttpRequest::redirected()
+{
+    flags.redirected = true;
```

Comment:
I suspect this method will go away completely and the new resetRequestXXX() will just set flags.redirected to true if needed (before calling checkForInternalAccess()).

resetRequestXXX() should document why it is important to set flags.redirected before calling checkForInternalAccess().

Reply:
> Probably you meant introducing some logical rule here

No, I did not. IIRC, we did discuss some flag setting/checking order dependency, but it is likely that I misunderstood something -- I do not see flags.redirected dependencies in the affected code. If there is no order dependency, we do not need to document it, of course. I am resolving this change request.